src/modules/localization/src/wheel_odom/odometry.cpp

```diff
@@ -122,24 +158,39 @@ namespace wheel_odometry_node
     }
 
     void WheelOdometryNode::odometry_state_estimation()
-    {
-        // bicycle model dead-reckoning
-        
+    {        
         // velocity magnitude estimate
         vel_magnitude = (left_angular_vel+right_angular_vel)*WHEEL_RADIUS/2; // [m/s]
         // calculating median of the velocities
         wheel_odom_median_filter();
         
+        ////////////////////////
+        // VELOCITY ESTIMATE //
+        ///////////////////////
+
         // yaw estimate from odometry and steering
-        yaw_estimate = previous_yaw + (vel_magnitude/WHEELBASE)*tan(steering_angle)*DT; // [rad]
-        previous_yaw = yaw_estimate;
+        // yaw_estimate = previous_yaw + (vel_magnitude/WHEELBASE)*tan(steering_angle)*DT; // [rad]
+        // previous_yaw = yaw_estimate;
         
         // velocity estimate in x and y
         x_velocity = vel_magnitude * cos(yaw_estimate); // [m/s]
         y_velocity = vel_magnitude * sin(yaw_estimate); // [m/s]
+
+        // twist messages
+        velocity_estimate.twist.linear.x = x_velocity; // [m/s]
+        velocity_estimate.twist.linear.y = y_velocity; // [m/s]
         
-        /*
-        TODO: Possibly add position dead-reckoning from wheel odometry
+        // velocity covariances 
+        float x_vel_covariance = 0.0021878/2; // [m^2/s^2]

```

Comment:
Would be nice to have one or two short sentences explaining the reasoning of the values.

Reply:
Sounds good